Assets/Scripts/MoveControl.cs

```diff
@@ -0,0 +1,71 @@
+using UnityEngine;
+using UnityEngine.UIElements;
+
+public class MoveControl : MonoBehaviour
+{
+    [SerializeField] Rigidbody2D rb2;
+    [SerializeField] float speed = 5f;
+    [SerializeField] Doodler doodler;
+    float direction = 0;
+
+    private Camera mainCamera;
+    private float cameraWidth;
+    private float cameraLeftBorder;
+    private float cameraRightBorder;
+    private float borderOffset = 1f;
+
+
+
+    private void Start()
+    {
+        mainCamera = Camera.main;
```

Comment:
Nice!
I like this cashing

I'd add possibility to choose a camera we are going to check from the editor by adding SerializedField to it